oak-query-spi/src/main/java/org/apache/jackrabbit/oak/spi/query/QueryIndex.java

```diff
@@ -415,7 +433,33 @@ public Builder setDelayed(boolean isDelayed) {
                 return this;
             }
 
+            /**
+             * @deprecated use {@link #addAdditionalMessage(String level, String s)} instead
+             * */
+            @Deprecated(forRemoval = true)
             public Builder addAdditionalMessage(Level level, String s) {
+                log.warn("use of deprecated API - this method is going to be removed in future Oak releases - see OAK-10768 for details");
```

Comment:
same as above: we should add the name of the new API.

Reply:
IMHO not necessary...